src/DistributedFFT.chpl

```diff
@@ -212,195 +215,156 @@ prototype module DistributedFFT {
-          c_memcpy(c_ptrTo(myplane[0,iy,z0]),
-                   c_ptrTo(src.localAccess[xSrc.first,iy,z0]),
-                   myLineSize);
+      for ix in xSrc {
+        // Y-transform
+        forall myzRange in batchedRange(zSrc, numTasks) {
+          ref elt = myplane[0, ySrc.first, myzRange.first];
+          if myzRange.size == batchSizeSm then yPlanSm.execute(elt);
+                                          else yPlanLg.execute(elt);
         }
 
-        for ix in xSrc {
-          // y-transform
-          const y0 = ySrc.first;
-          forall myzRange in batchedRange(zSrc) with (ref plan_y, ref plan_y1) {
-            select myzRange.size {
-                when numTransforms do plan_y.execute(myplane[0,y0,myzRange.first]);
-                when numTransforms+1 do plan_y1.execute(myplane[0,y0,myzRange.first]);
-                otherwise halt("Bad myzRange size");
-              }
-          }
-
-          // z-transform
-          // Offset to reduce collisions
-          const offset = (ySrc.size/numLocales)*here.id;
-          forall iy in 0.. #ySrc.size with (ref plan_z) {
-            const iy1 = (iy + offset)%ySrc.size + y0;
-            plan_z.execute(myplane[0,iy1,z0]);
-            // This is the transpose step
-            remotePut(dest[iy1,ix,z0], myplane[0,iy1,z0], myLineSize);
-            // If not last slice, copy over
-            if (ix != xSrc.last) {
-              c_memcpy(c_ptrTo(myplane[0,iy1,z0]),
-                       c_ptrTo(src.localAccess[ix+1,iy1,z0]),
-                       myLineSize);
-            }
+        // Z-transform, offset to reduce comm congestion/collision
+        forall iy in offset(ySrc) {
+          zPlan.execute(myplane[0, iy, zSrc.first]);
+          // This is the transpose step
+          copy(Dst[iy, ix, zSrc.first], myplane[0, iy, zSrc.first], myLineSize);
+          // If not last slice, copy over
+          if (ix != xSrc.last) {
+            copy(myplane[0, iy, zSrc.first], Src[ix+1, iy, zSrc.first], myLineSize);
           }
         }
+      }
 
-        // Wait until all communication is complete
-        allLocalesBarrier.barrier();
-
-        // x-transform
-        const x0 = xDest.first;
-        forall myzRange in batchedRange(zSrc) with (ref plan_x, ref plan_x1) {
-          for iy in yDest {
-            ref elt = dest.localAccess[iy, x0, myzRange.first];
-            select myzRange.size {
-                when numTransforms do plan_x.execute(elt);
-                when numTransforms+1 do plan_x1.execute(elt);
-                otherwise halt("Bad myzRange size");
-              }
-          }
-        }
+      // Wait until all communication is complete
+      allLocalesBarrier.barrier();
 
-        // End of on-loc
+      // X-transform
+      forall myzRange in batchedRange(zSrc, numTasks) {
+        for iy in yDst {
+          ref elt = Dst[iy, xDst.first, myzRange.first];
+          if myzRange.size == batchSizeSm then yPlanSm.execute(elt);
```

Comment:
These should be `xPlan*`. 

And you clearly use better variable names than I do! (At least, all my variables are not `x`,`xx`,`xxx`).

Reply:
I believe the code still worked, because for the NPB benchmark, `yPlan` and `xPlan` are the same.

Reply:
Oh, shoot -- good catch

Reply:
Fixed